extensions/amp-next-page/0.1/config.js

```diff
@@ -95,13 +90,23 @@ function assertSelectors(selectors) {
 /**
  * @param {!Element} context
  * @param {*} reco
- * @param {string} origin
- * @param {string=} sourceOrigin
+ * @param {string} documentUrl
  */
-function assertReco(context, reco, origin, sourceOrigin) {
+function assertReco(context, reco, documentUrl) {
+  user().assertString(reco.ampUrl, 'ampUrl must be a string');
+
+  // Rewrite relative URLs to absolute, relative to the canonical URL.
+  const isRelativeUrl = !/^(?:[a-z]+:)?\/\//.test(reco.ampUrl);
+  if (isRelativeUrl) {
+    const {canonicalUrl} = Services.documentInfoForDoc(context);
```

Comment:
the base for resolveRelativeUrl can just be `const base = getSourceUrl(documentUrl);` `canonicalUrl` is the non-AMP version of the AMP page and may actually be on a separate domain than AMP.

Reply:
My bad. Fixed.